dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -680,6 +685,53 @@ async fn cmd_db_sleds(
     Ok(())
 }
 
+#[derive(Tabled)]
+#[tabled(rename_all = "SCREAMING_SNAKE_CASE")]
+struct CustomerInstanceRow {
+    id: Uuid,
+    state: String,
+    propolis_id: Uuid,
+    sled_id: Uuid,
+}
+
+impl From<Instance> for CustomerInstanceRow {
+    fn from(i: Instance) -> Self {
+        CustomerInstanceRow {
+            id: i.id(),
+            state: format!("{:?}", i.runtime_state.state.0),
+            propolis_id: i.runtime_state.propolis_id,
+            sled_id: i.runtime_state.sled_id,
+        }
+    }
+}
+
+/// Run `omdb db instances`: list data about customer VMs.
+async fn cmd_db_instances(
+    datastore: &DataStore,
+    limit: NonZeroU32,
+) -> Result<(), anyhow::Error> {
+    use db::schema::instance::dsl;
+    let instances = dsl::instance
+        .limit(i64::from(u32::from(limit)))
+        .select(Instance::as_select())
+        .load_async(datastore.pool_for_tests().await?)
+        .await
+        .context("loading instances")?;
+
+    let ctx = || "listing instances".to_string();
+    check_limit(&instances, limit, ctx);
+
+    let rows = instances.into_iter().map(|i| CustomerInstanceRow::from(i));
```

Comment:
I like the way you did this, makes converting between things nice.
I'll think I'll go refactor some of the stuff I've done in this same file :)